File: my-code/1/main.js
// @Time     : 2021/05/28 17: 46
// @Author   : Ranshi
// @File     : 1. 两数之和.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
const twoSum = function(nums, target) {
	let map = new Map();
	for (let i in nums) {
		if (map.has(target - nums[i])) {
			return [map.get(target - nums[i]), i];
		} else {
			map.set(nums[i], i);
		}
	}
};

console.log(twoSum((nums = [2, 7, 11, 15]), (target = 9)));

File: my-code/1/main.go
package main

import "fmt"

// @Time     : 2021/08/03 19:49
// @Author   : Ranshi
// @File     : main.go

func twoSum(nums []int, target int) []int {
	numSet := make(map[int]int, len(nums))
	for i, v := range nums {
		if idx, ok := numSet[target-v]; ok {
			return []int{idx, i}
		}
		numSet[v] = i
	}
	return nil
}

func main() {
	fmt.Println(twoSum([]int{1, 2, 34, 5, 52, 3, 1, 4, 5, 6, 7}, 8))
}
